<a href="https://colab.research.google.com/github/swhacademy/tensorflow-guide/blob/master/0_4_KoNLPy_Mecab_%EC%84%A4%EC%B9%98.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Install KoNLPy in colaboratory
[KoNLPy 설치 문서](http://konlpy-ko.readthedocs.io/ko/v0.4.4/install/)

본 문서는 coaboratory에서 mecab을 설치하여 konlpy에서 사용하기 위해 만들었습니다.

This document was created for use in konlpy by installing mecab in coaboratory.

## install konlpy library

 Successfully installed konlpy-0.4.4

In [0]:
!pip install konlpy


    100% |████████████████████████████████| 19.4MB 2.8MB/s 
    100% |████████████████████████████████| 174kB 38.6MB/s 
  Stored in directory: /root/.cache/pip/wheels/0e/2b/e8/c0b818ac4b3d35104d35e48cdc7afe27fc06ea277feed2831a
Successfully built JPype1


## install jdk and install JPype1-py3

oracle jdk 설치 무방

In [0]:
!apt-get install openjdk-8-jdk-headless -qq > /dev/null
!pip3 install JPype1-py3

    100% |████████████████████████████████| 92kB 9.4MB/s 
  Stored in directory: /root/.cache/pip/wheels/76/7d/41/829109c81b0c4bdca8047fb9e0a573d4db3b671fbbe84a84f3
Successfully built JPype1-py3


## mecab-ko 설치
[mecab 설치 문서](https://bitbucket.org/eunjeon/mecab-ko)

In [0]:
import os
os.chdir('/tmp/')
!curl -LO https://bitbucket.org/eunjeon/mecab-ko/downloads/mecab-0.996-ko-0.9.1.tar.gz
!tar zxfv mecab-0.996-ko-0.9.1.tar.gz
os.chdir('/tmp/mecab-0.996-ko-0.9.1')
!./configure
!make
!make check
!make install


  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 1380k  100 1380k    0     0   907k      0  0:00:01  0:00:01 --:--:-- 1950k
mecab-0.996-ko-0.9.1/
mecab-0.996-ko-0.9.1/config.h.in
mecab-0.996-ko-0.9.1/Makefile.train
mecab-0.996-ko-0.9.1/ChangeLog
mecab-0.996-ko-0.9.1/configure
mecab-0.996-ko-0.9.1/swig/
mecab-0.996-ko-0.9.1/swig/version.h.in
mecab-0.996-ko-0.9.1/swig/Makefile
mecab-0.996-ko-0.9.1/swig/version.h
mecab-0.996-ko-0.9.1/swig/MeCab.i
mecab-0.996-ko-0.9.1/mkinstalldirs
mecab-0.996-ko-0.9.1/mecab-config.in
mecab-0.996-ko-0.9.1/configure.in
mecab-0.996-ko-0.9.1/mecab.iss.in
mecab-0.996-ko-0.9.1/aclocal.m4
mecab-0.996-ko-0.9.1/mecabrc.in
mecab-0.996-ko-0.9.1/INSTALL
mecab-0.996-ko-0.9.1/AUTHORS
mecab-0.996-ko-0.9.1/example/
mecab-0.996-ko-0.9.1/example/example_lattice.c
mecab-0.996-ko-0.9.1/

## automake 설치

In [0]:
!apt-get install automake

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  libnvidia-common-410
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  autoconf autotools-dev libsigsegv2 m4
Suggested packages:
  autoconf-archive gnu-standards autoconf-doc libtool gettext m4-doc
The following NEW packages will be installed:
  autoconf automake autotools-dev libsigsegv2 m4
0 upgraded, 5 newly installed, 0 to remove and 6 not upgraded.
Need to get 1,082 kB of archives.
After this operation, 3,994 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 libsigsegv2 amd64 2.12-1 [14.7 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 m4 amd64 1.4.18-1 [197 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic/main amd64 autoconf all 2.69-11 [322 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 autot

## automake install error
configure: error: no acceptable m4 could be found in $PATH.

Glibc 2.9 - 2.12 and GNU M4 1.4.11 - 1.4.15 have another strstr bug.

- install GNU M4 1.4.9

- install autoconf-2.69



In [0]:
import os
os.chdir('/tmp/')

 
!wget -O m4-1.4.9.tar.gz http://ftp.gnu.org/gnu/m4/m4-1.4.9.tar.gz
!tar -zvxf m4-1.4.9.tar.gz
os.chdir('/tmp/m4-1.4.9')
!./configure
!make
!make install

!apt-get install autoconf

--2019-04-24 12:19:42--  http://ftp.gnu.org/gnu/m4/m4-1.4.9.tar.gz
Resolving ftp.gnu.org (ftp.gnu.org)... 209.51.188.20, 2001:470:142:3::b
Connecting to ftp.gnu.org (ftp.gnu.org)|209.51.188.20|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 770647 (753K) [application/x-gzip]
Saving to: ‘m4-1.4.9.tar.gz’

m4-1.4.9.tar.gz     100%[===================>] 752.58K  1.44MB/s    in 0.5s    

2019-04-24 12:19:42 (1.44 MB/s) - ‘m4-1.4.9.tar.gz’ saved [770647/770647]

m4-1.4.9/
m4-1.4.9/acinclude.m4
m4-1.4.9/aclocal.m4
m4-1.4.9/AUTHORS
m4-1.4.9/BACKLOG
m4-1.4.9/bootstrap
m4-1.4.9/c-boxes.el
m4-1.4.9/ChangeLog
m4-1.4.9/checks/
m4-1.4.9/checks/001.comments
m4-1.4.9/checks/002.input_proc
m4-1.4.9/checks/003.input_proc
m4-1.4.9/checks/004.inhibiting
m4-1.4.9/checks/005.inhibiting
m4-1.4.9/checks/006.inhibiting
m4-1.4.9/checks/007.inhibiting
m4-1.4.9/checks/008.inhibiting
m4-1.4.9/checks/009.inhibiting
m4-1.4.9/checks/010.macro_argu
m4-1.4.9/checks/011.macro_argu
m4-1.4.9/chec

## install mecab-ko-dic
[mecab-ko-dic 설치 문서](https://bitbucket.org/eunjeon/mecab-ko-dic)

In [0]:
import os
os.chdir('/tmp')
!curl -LO https://bitbucket.org/eunjeon/mecab-ko-dic/downloads/mecab-ko-dic-2.0.1-20150920.tar.gz
!tar -zxvf mecab-ko-dic-2.0.1-20150920.tar.gz
os.chdir('/tmp/mecab-ko-dic-2.0.1-20150920')
!./autogen.sh
!./configure
!make
# !sh -c 'echo "dicdir=/usr/local/lib/mecab/dic/mecab-ko-dic" > /usr/local/etc/mecabrc'
!make install

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 46.3M  100 46.3M    0     0  2935k      0  0:00:16  0:00:16 --:--:-- 2581k
mecab-ko-dic-2.0.1-20150920/
mecab-ko-dic-2.0.1-20150920/configure.ac
mecab-ko-dic-2.0.1-20150920/Person-actor.csv
mecab-ko-dic-2.0.1-20150920/IC.csv
mecab-ko-dic-2.0.1-20150920/model.def
mecab-ko-dic-2.0.1-20150920/user-dic/
mecab-ko-dic-2.0.1-20150920/user-dic/place.csv
mecab-ko-dic-2.0.1-20150920/user-dic/README.md
mecab-ko-dic-2.0.1-20150920/user-dic/person.csv
mecab-ko-dic-2.0.1-20150920/user-dic/nnp.csv
mecab-ko-dic-2.0.1-20150920/Place.csv
mecab-ko-dic-2.0.1-20150920/rewrite.def
mecab-ko-dic-2.0.1-20150920/tools/
mecab-ko-dic-2.0.1-20150920/tools/add-userdic.sh
mecab-ko-dic-2.0.1-20150920/tools/mecab-bestn.sh
mecab-ko-dic-2.0.1-20150920/tools/convert_for_using_store.sh

## ldconfig
libmecab.so.2를 찾을 수 없는 에러가 나는 경우

실행 후 재시도



In [0]:
os.chdir('/tmp/mecab-ko-dic-2.0.1-20150920')
!ldconfig
!ldconfig -p | grep /usr/local/lib


	libmkl_vml_mc3.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_mc3.so
	libmkl_vml_mc2.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_mc2.so
	libmkl_vml_mc.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_mc.so
	libmkl_vml_def.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_def.so
	libmkl_vml_cmpt.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_cmpt.so
	libmkl_vml_avx512_mic.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_avx512_mic.so
	libmkl_vml_avx512.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_avx512.so
	libmkl_vml_avx2.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_avx2.so
	libmkl_vml_avx.so (libc6,x86-64) => /usr/local/lib/libmkl_vml_avx.so
	libmkl_tbb_thread.so (libc6,x86-64) => /usr/local/lib/libmkl_tbb_thread.so
	libmkl_sequential.so (libc6,x86-64) => /usr/local/lib/libmkl_sequential.so
	libmkl_scalapack_lp64.so (libc6,x86-64) => /usr/local/lib/libmkl_scalapack_lp64.so
	libmkl_scalapack_ilp64.so (libc6,x86-64) => /usr/local/lib/libmkl_scalapack_ilp64.so
	libmkl_rt.so (libc6,x86-6

## install mecab-python
python3 기준입니다.

In [0]:
# install mecab-python
import os
os.chdir('/content')

!git clone https://bitbucket.org/eunjeon/mecab-python-0.996.git
os.chdir('/content/mecab-python-0.996')

!python3 setup.py build
!python3 setup.py install


Cloning into 'mecab-python-0.996'...
remote: Counting objects: 17, done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 17 (delta 3), reused 0 (delta 0)
Unpacking objects: 100% (17/17), done.
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.6
copying MeCab.py -> build/lib.linux-x86_64-3.6
running build_ext
building '_MeCab' extension
creating build/temp.linux-x86_64-3.6
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/include -I/usr/include/python3.6m -c MeCab_wrap.cxx -o build/temp.linux-x86_64-3.6/MeCab_wrap.o
x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.6/MeCab_wrap.o -L/usr/local/lib -lmecab -lstdc

## complete

설치가 모두 완료되었습니다.

새창을 띄워 실행 해보시길 바랍니다.

